tests/python/relax/test_op_nn.py

```diff
@@ -1044,5 +1085,421 @@ def test_cross_entropy_infer_struct_info_wrong_input_type():
         bb.normalize(relax.op.nn.softmax_cross_entropy(x1, y))
 
 
+def test_nll_loss_infer_struct_info():
```

Comment:
Emmm, but there is no restriction for ignore_index. In InferStructInfo I don't even use its value.

Reply:
One check we can do in the inference is that, in case `ignore_index` is positive and we know what `C`, we can throw error if `ignore_index` is proven to be not less than `C`.